Sources/TuistKit/Mappers/Graph/StaticXCFrameworkModuleMapGraphMapper.swift

```diff
@@ -0,0 +1,182 @@
+                    for: staticObjcXCFrameworksLinkedByDynamicXCFrameworkDependencies,
+                    derivedDirectory: derivedDirectory
+                )
+
+                settingsToPropagate[.target(name: target.name, path: project.path)] = [
+                    "OTHER_SWIFT_FLAGS": .array(
+                        staticObjcXCFrameworksLinkedByDynamicXCFrameworkDependencies.flatMap { xcframework -> [String] in
+                            [
+                                "-Xcc",
+                                moduleMapFlag(
+                                    for: xcframework,
+                                    derivedDirectory: derivedDirectory,
+                                    project: project
+                                ),
+                            ]
+                        }
+                    ),
+                    "OTHER_C_FLAGS": .array(
+                        staticObjcXCFrameworksLinkedByDynamicXCFrameworkDependencies.flatMap { xcframework -> [String] in
+                            [
+                                moduleMapFlag(
+                                    for: xcframework,
+                                    derivedDirectory: derivedDirectory,
+                                    project: project
+                                ),
+                            ]
+                        }
+                    ),
+                    "HEADER_SEARCH_PATHS": .array(
+                        staticObjcXCFrameworksLinkedByDynamicXCFrameworkDependencies.flatMap { xcframework -> [String] in
+                            guard let moduleMap = xcframework.path.glob("**/module.modulemap").first
+                            else { return [] }
+                            return [
+                                "\"$(SRCROOT)/\(moduleMap.parentDirectory.relative(to: project.path).pathString)\"",
+                            ]
+                        }
+                    ),
+                ]
+            }
+        }
+
+        return (
+            try propagateSettings(graph: graph, settings: settingsToPropagate),
+            sideEffects,
+            environment
+        )
+    }
+
+    private func moduleMapFlag(
+        for xcframework: GraphDependency.XCFramework,
+        derivedDirectory: AbsolutePath,
+        project: Project
+    ) -> String {
+        let name = xcframework.path.basenameWithoutExt
+        return "-fmodule-map-file=\"$(SRCROOT)/\(derivedDirectory.appending(components: name, "Headers", "module.modulemap").relative(to: project.path).pathString)\""
+    }
+
+    private func derivedDirectory(graph: Graph) -> AbsolutePath {
+        let packageDirectoryPath: AbsolutePath
+        if fileHandler.exists(graph.path.appending(component: Constants.SwiftPackageManager.packageSwiftName)) {
+            packageDirectoryPath = graph.path
+        } else {
+            packageDirectoryPath = graph.path.appending(component: Constants.tuistDirectoryName)
+        }
```

Comment:
Is this the only place in the codebase where we need to get the `.build` directory based on the location of the `Package.swift`? If not, I'd consider making this logic reusable to prevent inconsistencies.